services/device-registry-base/src/main/java/org/eclipse/hono/service/management/device/Filter.java

```diff
@@ -0,0 +1,109 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.hono.service.management.device;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.eclipse.hono.util.RegistryManagementConstants;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import io.vertx.core.json.pointer.JsonPointer;
+
+/**
+ * Filter to apply during search operation in Device Registry Management API.
+ *
+ * @param <T> The filter value type.
+ */
+@JsonInclude(value = JsonInclude.Include.NON_NULL)
+public final class Filter<T> {
+
+    @JsonProperty(RegistryManagementConstants.FIELD_FILTER_FIELD)
+    private JsonPointer field;
+
+    @JsonProperty(RegistryManagementConstants.FIELD_FILTER_VALUE)
+    private T value;
+
+    @JsonProperty(RegistryManagementConstants.FIELD_FILTER_OPERATOR)
+    private OPERATOR operator = OPERATOR.eq;
+
+    /**
+     * An enum defining supported filter operators.
+     */
+    public enum OPERATOR {
+        eq
+    }
+
```

Comment:
can we add a constructor accepting the mandatory properties and remove *setField* and *setValue*?